hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableRatisContainerProvider.java

```diff
@@ -123,8 +119,25 @@ public ContainerInfo getContainer(final long size,
 
         } catch (SCMException se) {
           LOG.warn("Pipeline creation failed for repConfig {} " +
-              "Datanodes may be used up.", repConfig, se);
-          break;
+              "Datanodes may be used up. Try to see if any pipeline is in " +
+                  "ALLOCATED state, and then will wait for it to be OPEN",
+                  repConfig, se);
+          List<Pipeline> allocatedPipelines = findPipelinesByState(repConfig,
```

Comment:
It doesn't appear as if the test code covers waiting for pipelines to open.  Does it need to?

Reply:
Do you mean testing waitOnePipelineReady() method? This method involves the timer and multi thread, so I think it is not a good candidate for unit test. For integration test, I do not know how to set it up and test it through. Any suggestion?

Reply:
@sokui   i was thinking of something like this:
https://github.com/sokui/ozone/compare/HDDS-5916-support-datanode-change-ip-hostname...GeorgeJahad:gbjAllocateTest2

Feel free to ignore it if you don't like it.

Reply:
I usually do not use sleep() in unit tests, because it may cause the unit test unreliable. But for this test, we have no other better way to do it, and the sleep() here should not make the test case unstable. Let me include your commit. Thank you!